doc/troubleshooting.md

```diff
@@ -24,6 +24,68 @@ On Windows the `.cmd` file changes the `CWD` to the location of the script befor
+2. Modify the robot controller's IP and subnet mask so it is on the subnet of the PC running the micro-ROS agent.
+3. Modify the robot controller's network settings to add a gateway which can reach the Agent's IP address.
+
+Refer to the relevant Yaskawa Motoman documentation for information on how to change the controller's network configuration.
+
+Test if the two are connected by opening a terminal on the PC and attempting to ping the robot controller's IP. For example
+
+```shell
+ping 192.168.1.31
+```
+
+If you receive a near-immediate response upon attempting to ping the controller, then the potential network issues mentioned above are resolved. Try to connect again via the micro-ROS agent. If the micro-ROS agent still cannot connect to the controller, there may be a firewall configuration issue.
+
+If you do not receive a near-immediate after attempting to ping the controller, there is still some issue. You may have made a mistake previously, or there could be a firewall issue, or there could be a network hardware issue.
+
+### Firewall issues
+
+Firewall rules on the PC running the micro-ROS agent application may prevent it from connecting to the robot controller. The controller and the PC must be able to send UDP packets back and forth.
+
+Assuming the PC is running Ubuntu, you can check the status of the firewall using the default firewall configuration tool `ufw` with the command below.
+
+```shell
+sudo ufw status
+```
+
+If the status shows as active, you should use the following command to see what firewall rules have been added that may prevent communication.
+
+```shell
+sudo ufw show added
+```
+
+Look at the rules and remove any that may interfere with the connection for MotoROS2. For example, if the rule `ufw deny 8888/udp` was a present, you would want to remove it.
+
+Using `ros2 multicast send` and `ros2 multicast receive` can be helpful for diagnosing firewall problems.
+
+For further discussion on how to fix firewall issues, see [here](https://stackoverflow.com/a/75087882).
+
+If you wish to disable your firewall to test if it is the source of your problem, you may do so, but be aware of the risks associated with that.
+
+It is possible that firewall rules are not visible via `ufw`, but that they are still in place. That goes beyond the scope of this troubleshooting guide, but it is a possibility.
+
+### Other potential network problems
+
+There are other potential network configuration problems.
+
+Make sure that there are no IP address conflicts with either the client PC or the robot controller.
+
+Make sure that all of the connections between the controller and the PC are solid, and that all of the cables and peripheral equipment works properly.
+
+If all else fails, Wireshark can be a valuable tool for finding where network problems lie.
```

Comment:
I don't know that we should suggest this without providing some concrete definition of what to look for.